### PR TITLE
Added REST tags endpoint

### PR DESCRIPTION
**Pull request by nielstholenaar**
See  https://github.com/flarum/core/issues/600 for more information.

### CODE REVIEW

File: src/TagRepository.php [hunk 1]
```diff
@@ -17,6 +17,16 @@
 class TagRepository
 {
     /**
+     * Get a new query builder for the users table.
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function query()
+    {
+        return Tag::query();
```
Comment (franzliedke) [inline src/TagRepository.php]: Why is this `query` when we used `newQuery` below?

**Comment by franzliedke** [comment]
This is looking very good. Just one little question.

File: src/TagRepository.php (hunk 1 above)
Comment (nielstholenaar) [inline src/TagRepository.php]: Because the tag model doesn't has an newQuery method.

**Comment by nielstholenaar** [comment]
I've chosen to use the `Tag::query()` method instead of the `Tag::newQuery()`.

Mainly because the `Tag::query()


The newQuery method errors out with the message: `Call to undefined method Flarum\Tags\TagRepository::newQueryWithoutScopes()`

**Comment by tobscure** [comment]
Also ListTagController should be ListTag**s**Controller :)

**Comment by nielstholenaar** [comment]
@tobscure I've renamed it :-)

**Comment by tobscure** [comment]
Awesome! Oh, I just noticed that this will spit out all tags regardless of permissions. I think `$this->tags->all($request->getAttribute('actor'));` should work? If you can test it out a bit that'd be great :)

**Comment by franzliedke** [comment]
I've done that now.

Thanks so much @nielstholenaar!

**Pull request merged**